nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/AbstractMongoProcessor.java

```diff
@@ -221,4 +264,16 @@ protected WriteConcern getWriteConcern(final ProcessContext context) {
         }
         return writeConcern;
     }
+
+    protected void writeBatch(String payload, FlowFile parent, ProcessContext context, ProcessSession session, Map extraAttributes, Relationship rel) throws UnsupportedEncodingException {
+        String charset = parent != null ? context.getProperty(CHARSET).evaluateAttributeExpressions(parent).getValue()
+                : context.getProperty(CHARSET).evaluateAttributeExpressions().getValue();
+
+        FlowFile flowFile = session.create(parent);
```

Comment:
Need to check if parent is null before using it in the create. A unit test where there is no incoming flow file will illustrate this.

Reply:
I just threw a ternary operator in there.

Reply:
The ternary operator for charset is fine, I'm referring to the session.create(parent). If parent is null, then an NPE will be thrown when it tries to get the parent's attributes.